android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/WebViewFragment.java

```diff
@@ -137,7 +137,7 @@ public void onPageFinished(WebView view, String url) {
         // Inject a meta viewport tag into the head of the file if it doesn't exist
         webView.loadUrl(
           "javascript:(function() {" +
-            "if(!document.querySelectorAll('meta[name=viewport]').length) {"+
+            "if(document.head && !document.querySelectorAll('meta[name=viewport]').length) {"+
```

Comment:
If the page was lagging enough, this would throw a console error at l. 144 about `appendChild()` failing

Reply:
Hmm.  So there's no easy way to detect this scenario and retry later?

Maybe we could adapt this into a keyboard.html function that sets a timeout or an event against `document.readyState` (see https://developer.mozilla.org/en-US/docs/Web/API/Document/readystatechange_event) to call this when it _is_ possible to handle?

Reply:
I think I'll defer this to see how you implement #4555

Reply:
That sounds fine to me; @darcywong00 can you open an issue on that? I think we should do the same pattern for iOS and Android.

Reply:
> can you open an issue on that?

Android will implement in #4556. I'll make a copy for iOS